pyomo/contrib/preprocessing/plugins/remove_zero_terms.py

```diff
@@ -36,18 +39,18 @@ def _apply_to(self, model):
                 ctype=Constraint, active=True, descend_into=True):
             if not constr.body.polynomial_degree() == 1:
                 continue  # we currently only process linear constraints
-            repn = generate_canonical_repn(constr.body)
+            repn = generate_standard_repn(constr.body)
 
             # get the index of all nonzero coefficient variables
             nonzero_vars_indx = [
-                i for i, _ in enumerate(repn.variables)
-                if not repn.linear[i] == 0
+                i for i, _ in enumerate(repn.linear_vars)
+                if not isclose(repn.linear_coefs[i],0)
```

Comment:
This is a change in the behavior of the transformation and shouldn't be part of expr_dev.  If you think the transformation should use `isclose()` then there should be a ticket and the contrib package owner should make that call.

Reply:
Reverted...